`TestWriteMovieCanBeCanceled` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/6010599105/job/16302356389#step:6:1154

#### Relevant snippet

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xed37ef8]

goroutine 130 [running]:
testing.tRunner.func1.2({0xfb7df80, 0x187b8dc0})
make: *** [Makefile:754: test-go-tsh] Error 1
	/opt/go/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/opt/go/src/testing/testing.go:1529 +0x650
panic({0xfb7df80, 0x187b8dc0})
	/opt/go/src/runtime/panic.go:890 +0x263
github.com/gravitational/teleport/tool/tsh/common.writeMovie({0x136466c0, 0xc004742050}, {0x13643b80, 0xc007cd21a0}, {0x10baafb9, 0x4}, {0x10c13570, 0x8})
	/__w/teleport/teleport/tool/tsh/common/recording_export.go:187 +0xf78
github.com/gravitational/teleport/tool/tsh/common.TestWriteMovieCanBeCanceled(0x0?)
	/__w/teleport/teleport/tool/tsh/common/recording_export_test.go:58 +0x1d2
testing.tRunner(0xc0020421a0, 0x11150e10)
	/opt/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
	/opt/go/src/testing/testing.go:1629 +0x806
```

https://github.com/gravitational/teleport/blob/1908b611b5c3f3f85f9c2e6821971e2acc6db921/tool/tsh/common/recording_export.go#L187

--- a/tool/tsh/common/recording_export.go
+++ b/tool/tsh/common/recording_export.go
@@ -184,6 +184,12 @@ loop:
 		}
 	}
 
+	// if we received a session start event but the context is canceled
+	// before we received the screen dimensions, then there's no movie to close
+	if movie == nil {
+		return 0, trace.BadParameter("operation canceled")
+	}
+
 	return frameCount, trace.Wrap(movie.Close())
 }
 

